pmbootstrap: switch to pyproject

crftbt opened the pull request:
Update pmbootstrap from 2.2.0 to 2.3.1.

## Description of changes

https://gitlab.com/postmarketOS/pmbootstrap/-/tags/2.3.0
and
https://gitlab.com/postmarketOS/pmbootstrap/-/tags/2.3.1

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

crftbt committed:
pmbootstrap: 2.2.0 -> 2.3.1

Update pmbootstrap from 2.2.0 to 2.3.1.

getchoo reviewed (changes requested):
as of 2.3.0

>  We've migrated from setup.py to pyproject.toml. If you're packaging
   pmbootstrap, you may have to adjust the build recipe.
   
builds are failing due to this

onny commented:
Result of `nixpkgs-review pr 312969` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages failed to build:</summary>
  <ul>
    <li>pmbootstrap</li>
    <li>pmbootstrap.dist</li>
  </ul>
</details>

crftbt committed:
pmbootstrap: 2.2.0 -> 2.3.1

Updating buildPythonApplication to python3Packages and adding nativeBuildInputs for setuptools

onny commented:
please squash merge the two commits into one or change the commit title of one of them to: `pmbootstrap: switch to pyproject`

thank you :)

crftbt commented:
> please squash merge the two commits into one or change the commit title of one of them to: `pmbootstrap: switch to pyproject`
> 
> thank you :)

The title change makes sense. It's updated here for me "pmbootstrap: switch to pyproject" https://github.com/NixOS/nixpkgs/pull/312969/commits/e889a40276b9063055a378b7865efd70a1194d73. But I still see the old title on https://github.com/NixOS/nixpkgs/pull/312969/commits. Is it showing properly for you?

lucasew commented:
@ofborg eval

MaeIsBad commented:
I think this can be closed considering https://github.com/NixOS/nixpkgs/pull/315082 was merged

lucasew commented:
> I think this can be closed considering #315082 was merged

I think so. The original maintainer was the one that merged.

Thank you for the PR tho
